core/arxiv/submission/process/checks/tex_produced/__init__.py

```diff
@@ -0,0 +1,354 @@
+            f, e = walk(obj['/Resources'], fonts, embedded)
+            fonts = fonts.union(f)
+            embedded = embedded.union(e)
+
+        # I don't believe we need the embedded fonts but I'm leaving
+        # it here for now. This might be useful when we check whether a file
+        # embeds standard system fonts.
+        #unembedded = fonts - embedded
+
+    except utils.PdfReadError as ex:
+        logger.error(f"get_pdf_fonts_from_stream: Can't read PDF: Error: {ex}")
+        raise ex
+    except Exception as ex:
+        logger.error(f"Error: {ex}")
+        raise ex
+
+    return sorted(list(fonts))
+
+# Regexes to help us parse list of fonts from pdffonts.
+
+regex_header = re.compile(br"^\-\-\-\-", re.IGNORECASE)
+regex_fnt = re.compile(br"^.*\+[^\ ]*|^.*\-[^\ ]*", re.IGNORECASE)
+
+
+def get_pdf_fonts_from_file(filepath: str) -> List[str]:
+    """
+    Return the list of fonts for specified file.
+
+    Parameters
+    ----------
+    filepath : str
+        The file to run pdffonts on.
+
+    Returns
+    -------
+    List containing one entry for each line of output from pdffonts command.
+
+    """
+    fonts_list = []
+
+    info = subprocess.run(["pdffonts", f"{filepath}"], stderr=subprocess.PIPE,
+                          stdout=subprocess.PIPE)
+
+    lines = info.stdout.splitlines()
+
+    # Clean up the output by extracting font names
+    for line in lines:
+        if regex_header.search(line):
+            continue
+        match = regex_fnt.search(line)
+        if match:
+            fonts_list.append(match[0])
+
+    return sorted(fonts_list)
+
+
+# Regular expressions that indicate TeX Produced (for pdfinfo)
+
+regex1 = re.compile(br"^Creator.*dvips", re.IGNORECASE)
+regex2 = re.compile(br"^Title.*\.dvi", re.IGNORECASE)
+regex3 = re.compile(br"^(Creator|Producer).*TeX(?!(t|macs-1\.))", re.IGNORECASE)
+regex4 = re.compile(br"^(Creator|Producer).*dvipdfmx?", re.IGNORECASE)
+regex5 = re.compile(br"^(Creator|Producer).*cairo.*$", re.IGNORECASE)
+
+regexes = [regex1, regex2, regex3, regex4, regex5]
+
+# Ignore this case where wisywig sotware uses TeX for back end processing.
+regexTeXmacs = re.compile(br"^(Creator|Producer).*TeXmacs-1\.", re.IGNORECASE)
+
+# PDFs using Cairo or CMR/RTX fonts were generated by TeX
+fontsrgex = re.compile(br"^CairoFont\-\d-\d|^\S+\+(?:cmr|rtx)", re.IGNORECASE)
+
+
+# PDF
+def check_tex_produced_pdf_from_stream(stream: IO[bytes]) -> bytes:
+    """
+    Check whether specified PDF file was produced by TeX.
+
+    Parameters
+    ----------
+    stream : IO[bytes]
+        The stream of file bytes for PDF to be checked for TeX produced.
+
+    Returns
+    -------
+        String (regex match) if PDF is TeX-produced, otherwise return ''.
+
+    """
+    info = []
+    try:
+        info = get_filtered_pdf_info_from_stream(stream)
+    except Exception as ex:
+        logger.error(f'Error getting pdfinfo: {ex}')
+
+    if verbose:
+        print(f"\nInfo: '{info}' from PDF")
```

Comment:
these `print` statements could be `logger.debug` calls

Reply:
I'll clean these up.

Reply:
Done